src/main/java/org/prebid/server/bidder/grid/GridKeywordsUtil.java

```diff
@@ -0,0 +1,244 @@
+package org.prebid.server.bidder.grid;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.apache.commons.lang3.StringUtils;
+import org.prebid.server.bidder.grid.model.KeywordSegment;
+import org.prebid.server.bidder.grid.model.Keywords;
+import org.prebid.server.bidder.grid.model.KeywordsPublisherItem;
+import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.util.ObjectUtil;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class GridKeywordsUtil {
+
+    private static final TypeReference<Map<String, JsonNode>> MAP_TYPE_REF =
+            new TypeReference<Map<String, JsonNode>>() {
+            };
+
+    private GridKeywordsUtil() {
+    }
```

Comment:
Mapper is provided as an argument in every method, let's create static declaration in this class for it